### PR TITLE
light: create provider options struct

### PR DESCRIPTION
**Pull request by cmwaters**
https://github.com/tendermint/tendermint/pull/6054#discussion_r571149135

Also adds the functionality that a negative maxRetryAttempts value will mean that the light client will query indefinitely

### CODE REVIEW

File: light/provider/http/http.go [hunk 1]
```diff
@@ -121,7 +130,7 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 	)
 
 	for len(vals) != total && page <= maxPages {
-		for attempt := 1; attempt <= p.maxRetryAttempts; attempt++ {
+		for attempt := 0; attempt <= p.maxRetryAttempts; attempt++ {
```
Comment (melekes) [inline light/provider/http/http.go:133]:
```suggestion
		for attempt := 0; attempt < p.maxRetryAttempts; attempt++ {
```

?

File: light/provider/http/http.go [hunk 2]
```diff
@@ -166,7 +175,7 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 }
 
 func (p *http) signedHeader(ctx context.Context, height *int64) (*types.SignedHeader, error) {
-	for attempt := 1; attempt <= p.maxRetryAttempts; attempt++ {
+	for attempt := 0; attempt <= p.maxRetryAttempts; attempt++ {
```
Comment (melekes) [inline light/provider/http/http.go:178]:
```suggestion
	for attempt := 0; attempt < p.maxRetryAttempts; attempt++ {
```

?

File: light/provider/http/http.go (hunk 1 above)
Comment (cmwaters) [reply light/provider/http/http.go:133]: but in that case if `maxRetryAttempts` is 0 then it won't even make the initial request

File: light/provider/http/http.go [hunk 3]
```diff
@@ -30,39 +32,46 @@ type http struct {
 	maxRetryAttempts int
 }
 
+type Options struct {
+	// -1 means no limit
```
Comment (melekes) [inline light/provider/http/http.go:36]: should we do this in a separate PR?

Comment (cmwaters) [reply light/provider/http/http.go:36]: This PR is probably small enough that we can fit it in. I just have to change `attempt <= p.maxRetryAttempts` to `attempt != p.maxReryAttempts`

**Review by melekes — APPROVED** [review]
:+1:
